### PR TITLE
[1/X] [DXCDT-175] Add support for webauthn in guardian resource

### PR DESCRIPTION
**Pull request by sergiughf**
## Description

Fixes partially: https://github.com/auth0/terraform-provider-auth0/issues/165, more to follow after this PR. Depends on https://github.com/auth0/go-auth0/pull/83. 
 

## Checklist

**Note:** Checklist required to be completed before a PR is considered to be reviewable.

#### Auth0 Code of Conduct
- [x] I have read and agreed to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)

#### Auth0 General Contribution Guidelines
- [x] I have read the [Auth0 General Contribution Guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)

#### Changes include test coverage?
- [x] Yes
- [ ] Not needed

#### Does the description provide the correct amount of context?
- [x] Yes, the description provides enough context for the reviewer to understand what these changes accomplish

#### Have you updated the documentation?
- [x] Yes, I've updated the appropriate docs
- [ ] Not needed

#### Is this code ready for production?
- [x] Yes, all code changes are intentional and no debugging calls are left over

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -54,8 +54,8 @@ github.com/apparentlymart/go-textseg/v12 v12.0.0/go.mod h1:S/4uRK2UtaQttw1GenVJE
 github.com/apparentlymart/go-textseg/v13 v13.0.0 h1:Y+KvPE1NYz0xl601PVImeQfFyEy6iT90AvPUL1NNfNw=
 github.com/apparentlymart/go-textseg/v13 v13.0.0/go.mod h1:ZK2fH7c4NqDTLtiYLvIkEghdlcqw7yxLeM89kiTRPUo=
 github.com/armon/go-socks5 v0.0.0-20160902184237-e75332964ef5/go.mod h1:wHh0iHkYZB8zMSxRWpUBQtwG5a7fFgvEO+odwuTv2gs=
-github.com/auth0/go-auth0 v0.7.0 h1:mw7KbtVfAdVhJqinPeCNQQRuP4kv7eXe2GujaULgPWs=
```
Comment (Widcket) [inline go.sum:57]: Should we use a feature branch for multi-PR work instead of leaving master in a non-releasable state?

**Review by Widcket — CHANGES_REQUESTED** [review]
Let's use a feature branch for multi-PR work.

File: go.sum (hunk 1 above)
Comment (sergiughf) [reply go.sum:57]: We won't be merging this until we update the go.mod to point to a correct stable release:) This is just to showcase that tests are passing with the work done in go-auth0.

Comment (Widcket) [reply go.sum:57]: So, should the PR be a draft PR?

Comment (sergiughf) [reply go.sum:57]: Actually yes, that definitely makes more sense 👍🏻

**Review by Widcket — APPROVED** [review]

**Pull request merged**